autohooks/api/path.py

```diff
@@ -14,34 +14,47 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Plugin API for path checking
+"""
 
 import fnmatch
 import os
 from os import PathLike
 from pathlib import Path
-from typing import Iterable
+from typing import Iterable, Optional
 
 
-def is_python_path(path: Path) -> bool:
+def is_python_path(path: Optional[Path]) -> bool:
+    """
+    Function to check if path is a Python file.
+
+    Args:
+        path: A path to a file.
+
+    Returns:
+        True if path is a Python file.
+    """
     if not path:
         return False
     return path.match("*.py")
```

Comment:
```suggestion
    if not path or not path.exists():
        return False
    return path.suffix == "*.py"
```

We could use `path.suffix` here (i know its not related to you changes ...)?

Reply:
Not sure if that really improves the code. Personally I like the match method because it fits to the other function in this module.

Reply:
oh and it needs to be `return path.suffix == ".py"`